source/jormungandr/jormungandr/olympic_site_params_manager.py

```diff
@@ -110,27 +54,55 @@ def build_olympic_site_params(self, scenario, data):
             for spt_id, d in data.get("scenarios", {}).get(scenario, {}).get("stop_points", {}).items()
         }
 
-    def get_dict_scenario(self, poi_uri, key):
+    def get_valid_scenario_name(self, scenario_list, key, datetime):
+        for event in scenario_list:
+            if event["start"] <= datetime <= event["end"]:
+                return event.get(key)
+        return None
+
+    def get_dict_scenario(self, poi_uri, key, datetime):
         data = self.olympic_site_params.get(poi_uri)
         if not data:
             return {}
-        return self.build_olympic_site_params(data.get(key), data)
+        scenario_name = self.get_valid_scenario_name(data.get("events", []), key, datetime)
+        if scenario_name:
+            return self.build_olympic_site_params(scenario_name, data)
+        return {}
 
-    def get_dict_additional_parameters(self, poi_uri, key):
+    def get_dict_additional_parameters(self, poi_uri, key, datetime):
         data = self.olympic_site_params.get(poi_uri)
         if not data:
             return {}
-        scenario = data.get(key)
-        if not scenario:
-            return {}
-        return data.get("scenarios", {}).get(scenario, {}).get("additional_parameters", {})
+        scenario_name = self.get_valid_scenario_name(data.get("events", []), key, datetime)
+        if scenario_name:
+            return data.get("scenarios", {}).get(scenario_name, {}).get("additional_parameters", {})
+        return {}
 
     def get_strict_parameter(self, poi_uri):
         data = self.olympic_site_params.get(poi_uri)
         if not data:
             return False
         return data.get("strict", False)
 
+    def get_timestamp(self, str_datetime):
+        if self.instance.timezone:
+            dt = local_str_date_to_utc(str_datetime, self.instance.timezone)
+        else:
+            dt = str_to_dt(str_datetime)
+        return date_to_timestamp(dt) if dt else None
+
+    def str_datetime_time_stamp(self, json_data):
+        """
+        transform from_datetime, to_datetime to time_stamp
+        """
+        for key, value in json_data.items():
+            for event in value.get("events", []):
+                str_from_datetime = event.get("from_datetime")
+                str_to_datetime = event.get("to_datetime")
+                if str_from_datetime and str_to_datetime:
+                    event["start"] = self.get_timestamp(str_from_datetime)
+                    event["end"] = self.get_timestamp(str_to_datetime)
```

Comment:
```suggestion
                    event["from_datetime_utc"] = self.get_timestamp(str_from_datetime)
                    event["to_datetime_utc"] = self.get_timestamp(str_to_datetime)
```
and rename everywhere else

Reply:
:+1: Done